docs/website/docs/reference/mlir-dialects/HAL.md

```diff
@@ -0,0 +1,5 @@
+[include "HALDialect.md"]
+
+## Passes
+
+[include "HALPasses.md"]
```

Comment:
Here's a screenshot of this rendered page (standalone, not included):
![image](https://github.com/user-attachments/assets/4287ed7a-4547-4ce6-8414-24292c335af0)

We'll want a table of contents at the top and may want to hide / move the `Options` subheadings down. https://github.com/iree-org/iree/blob/main/docs/website/postprocess_dialect_docs.py would be the place for such edits

Reply:
I added a table of contents on the passes page.
Each pass has its own specific options. This is how MLIR's tablegen generates the doc.

Reply:
> Each pass has its own specific options. This is how MLIR's tablegen generates the doc.

Yep. I was thinking that the `Options` in the table of contents were noisy. Looks like `heading-offset=1` that you used did what I wanted.

Without the heading offset: 
![image](https://github.com/user-attachments/assets/d8d35caa-84a6-4e0a-b887-47616f2915da)

With it:
![image](https://github.com/user-attachments/assets/f54d5140-f4ae-4b67-9022-34b60f5cbb32)